src/IIIFPresentation/BackgroundHandler/BatchCompletion/BatchCompletionMessageHandler.cs

```diff
@@ -184,11 +189,87 @@ private void UpdateCanvasPaintings(Batch batch, Dictionary<AssetId, Canvas> item
             canvasPainting.Thumbnail = thumbnailPath != null ? new Uri(thumbnailPath) : null;
             canvasPainting.Ingesting = false;
             canvasPainting.Modified = DateTime.UtcNow;
-            canvasPainting.StaticHeight = item.Height;
-            canvasPainting.StaticWidth = item.Width;
+
+            if (canvasPainting is {StaticWidth: null, StaticHeight: null})
+            {
+                // #232: set static width/height if not provided.
+                // if canvas painting comes with statics, use them
+                // otherwise, if we can find dimensions within, use those
+                // ReSharper disable once InvertIf
+                if (GetCanvasDimensions(item) is var (width, height))
+                {
+                    canvasPainting.StaticWidth = width;
+                    canvasPainting.StaticHeight = height;
+                }
+            }
+            else if (canvasPainting is {StaticWidth: { } staticWidth, StaticHeight: { } staticHeight}
+                     && item.Items?.GetFirstPaintingAnnotation()?.Body is Image image
+                     && manifest.SpaceId.HasValue) // required for ImageRequest parsing and modification
```

Comment:
I don't think `&& manifest.SpaceId.HasValue` is correct. 

I could submit X assets to a manifest and specify the space for all of them - in that case the manifest wouldn't have a space. 
OR I could submit X assets and all but 1 have a space so the manifest would have a space but it would only be relevant for the 1 without.

`canvasPainting.AssetId` is the best place to read space from, it's an `AssetId` construct which has `Space` as part of it - this class mimics what's used internally in Protagonist.

Reply:
Aaah there it is. Thanks - I'd have asked about that anyway today :D

Reply:
Switched to using `AssetId` values`